crates/circuit/src/dag_circuit.rs

```diff
@@ -5785,6 +5785,26 @@ impl DAGCircuit {
         }
     }
 
+    // Filter any nodes that don't match a given predicate function
+    pub fn filter_op_nodes<F, E>(&mut self, mut predicate: F) -> Result<(), E>
+    where
+        F: FnMut(&PackedInstruction) -> Result<bool, E>,
```

Comment:
I wonder if we should allow the callable to be fallible at all. I know we have often supported this in Rustworkx core's API to make it easier to use with the wild world of Python. But in this case, maybe it's okay to require the user to give us a function that isn't expected to fail.

Otherwise, we'll be asking them to define `Infallible` result types probably more often than not.

Reply:
I'm fine with dropping the result here, I added it because when I originally wrote this I intended to use this method for the python function. But because that wasn't possible because of different interfaces between python and rust. So I agree this probably doesn't need to be fallible if we're not calling this with a python callback. I'll remove the result as it'll make it easier to work with.

Reply:
Done in: https://github.com/Qiskit/qiskit/pull/13052/commits/5e00d3c416172a82d63e0ca378ef2465292edd7e